bolts/fair/models/gpd.py

```diff
@@ -1,19 +1,21 @@
 """Zhang Gradient Projection Debiasing Baseline Model."""
-from typing import Dict, List, NamedTuple, Tuple
+from __future__ import annotations
+from typing import Mapping, NamedTuple
 
 import ethicml as em
 from kit import implements
 import pandas as pd
 import pytorch_lightning as pl
 import torch
 from torch import Tensor, nn, optim
-from torch.optim.lr_scheduler import CosineAnnealingWarmRestarts, _LRScheduler
+from torch.optim.lr_scheduler import CosineAnnealingWarmRestarts
 import torchmetrics
 from torchmetrics import MetricCollection
 
-from bolts.common import Stage
+from bolts.common import Stage, get_args
 from bolts.fair.data import DataBatch
 from bolts.fair.losses import CrossEntropy
+from bolts.fair.models import LRScheduler, SchedInterval
```

Comment:
this risks circular import otherwise
```suggestion
from bolts.fair.models.utils import LRScheduler, SchedInterval
```